utils.c

```diff
@@ -883,3 +884,82 @@ label_exec (UNUSED const char *exec_label)
 #endif
   return 0;
 }
+
+void
+strappend (StringBuilder *dest, const char *src)
+{
+  size_t len = strlen (src);
+
+  if (dest->offset + len >= dest->size)
+    {
+      dest->size = (dest->size + len + 1) * 2;
+      dest->str = xrealloc (dest->str, dest->size);
+    }
+
+  strncpy (dest->str + dest->offset, src, len + 1);
+  dest->offset += len;
+}
+
+__attribute__((format (printf, 2, 3)))
+void
+strappendf (StringBuilder *dest, const char *fmt, ...)
+{
+  va_list args;
+  int len;
+
+  va_start (args, fmt);
+  len = vsnprintf (dest->str + dest->offset, dest->size - dest->offset, fmt, args);
+  va_end (args);
+  if (len < 0)
+    die_with_error ("vsnprintf");
+  if (dest->offset + len >= dest->size)
+    {
+      dest->size = (dest->size + len + 1) * 2;
+      dest->str = xrealloc (dest->str, dest->size);
+      va_start (args, fmt);
+      len = vsnprintf (dest->str + dest->offset, dest->size - dest->offset, fmt, args);
+      va_end (args);
+      if (len < 0)
+        die_with_error ("vsnprintf");
+    }
+
+  dest->offset += len;
+}
+
+void
+strappend_escape_for_mount_options (StringBuilder *dest, const char *src)
+{
+  bool unescaped = TRUE;
+
+  for (;;)
+    {
+      if (dest->offset == dest->size)
+        {
+          dest->size = MAX (64, dest->size * 2);
```

Comment:
What is this max 64 here for?

Assuming we hit `dest->offset == dest->size == 64` the following switch will result in a buffer overflow.

Reply:
If this is the first function that's called on a `StringBuilder` this will also always allocate 0 btes.

Reply:
Did you perhaps mistake max for min? This ensures that the allocation is at least 64 bytes, not at most 64 bytes.

Reply:
woops, I did